lib/modules/manager/glasskube/extract.ts

```diff
@@ -0,0 +1,123 @@
+import { readLocalFile } from '../../../util/fs';
+import { parseYaml } from '../../../util/yaml';
+import { GlasskubePackagesDatasource } from '../../datasource/glasskube-packages';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFile,
+  PackageFileContent,
+} from '../types';
+import {
+  GlasskubeResource,
+  type Package,
+  type PackageRepository,
+} from './schema';
+import type { GlasskubeResources } from './types';
+
+function parseResources(
+  content: string,
+  packageFile: string,
+): GlasskubeResources {
+  const resources: GlasskubeResource[] = parseYaml(content, {
+    json: true,
+    customSchema: GlasskubeResource,
+    failureBehaviour: 'filter',
+  });
+
+  const packages: Package[] = [];
+  const repositories: PackageRepository[] = [];
+
+  for (const resource of resources) {
+    if (resource.kind === 'ClusterPackage' || resource.kind === 'Package') {
+      packages.push(resource);
+    } else if (resource.kind === 'PackageRepository') {
+      repositories.push(resource);
+    }
+  }
+
+  return { packageFile, repositories, packages };
+}
+
+function resolvePackageDependencies(
+  packages: Package[],
+  repositories: PackageRepository[],
+): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+  for (const pkg of packages) {
+    const dep: PackageDependency = {
+      depName: pkg.spec.packageInfo.name,
+      currentValue: pkg.spec.packageInfo.version,
+      datasource: GlasskubePackagesDatasource.id,
+    };
+
+    const repository = findRepository(
+      pkg.spec.packageInfo.repositoryName ?? null,
+      repositories,
+    );
+
+    if (repository === null) {
+      dep.skipReason = 'unknown-registry';
+    } else {
+      dep.registryUrls = [repository.spec.url];
+    }
+
+    deps.push(dep);
+  }
+  return deps;
+}
+
+function findRepository(
+  name: string | null,
+  repositories: PackageRepository[],
+): PackageRepository | null {
+  return (
+    repositories.find(
+      (it) =>
+        name === it.metadata.name ||
+        ((name === null || name === '') && isDefaultRepository(it)),
+    ) ?? null
+  );
```

Comment:
```suggestion
  for (const repository of repositories) {
    if (name === repository.metadata.name) {
      return repository
    }
    if (is.falsy(name) && isDefaultRepository(repository)) {
      return repository
    }
  }

  return null;
```
Small readability improvement, else LGTM